### PR TITLE
Use spatial low rank decomposition in low rank Trotter step

### LINKED ISSUE #254: Change low rank Trotter step to use spatial low rank routine
@jarrodmcc , @babbush any comments?

### PR DESCRIPTION
**Pull request by kevinsung**
Fixes #254 .

This change enabled the LiH test with truncated rank to work with rank 2 instead of 15. Random interaction operators don't work though.

### CODE REVIEW

File: openfermioncirq/trotter/algorithms/low_rank.py [hunk 1]
```diff
@@ -102,18 +101,16 @@ def __init__(self,
...
-        self.eigenvalues, self.one_body_squares, _ = (
-            low_rank_two_body_decomposition(
-                chemist_two_body_coefficients,
+        self.eigenvalues, self.one_body_squares, _, one_body_correction = (
+            low_rank_spatial_two_body_decomposition(
+                hamiltonian.two_body_tensor,
                 truncation_threshold=self.truncation_threshold,
-                final_rank=self.final_rank))
+                final_rank=self.final_rank,
+                spin_basis=True))
```
Comment (babbush) [inline openfermioncirq/trotter/algorithms/low_rank.py]: Shouldn't spin_basis = True be a default option for this function?

File: openfermioncirq/trotter/simulate_trotter_test.py [hunk 2]
```diff
@@ -35,40 +33,49 @@ def fidelity(state1, state2):
     return abs(numpy.dot(state1, numpy.conjugate(state2)))**2
 
 
+def load_molecular_hamiltonian(
```
Comment (babbush) [inline openfermioncirq/trotter/simulate_trotter_test.py]: why is this function in OpenFermion-Cirq. It should definitely not be in this library, let alone this module. Please remove.

File: openfermioncirq/trotter/simulate_trotter_test.py [hunk 3]
```diff
@@ -78,55 +85,67 @@ def produce_simulation_test_parameters(
     # Make sure the time is not too small
     assert fidelity(exact_state, initial_state) < .95
 
-    return hamiltonian, initial_state, exact_state
+    return initial_state, exact_state
+
 
+# Produce test parameters
 
+bigger_time = 1.0
```
Comment (babbush) [inline openfermioncirq/trotter/simulate_trotter_test.py]: longer_time?

File: openfermioncirq/trotter/simulate_trotter_test.py (hunk 2 above)
Comment (kevinsung) [reply openfermioncirq/trotter/simulate_trotter_test.py]: It's used to load test Hamiltonians (see below). I'm happy to put it somewhere in OpenFermion that you think is appropriate.

Comment (babbush) [reply openfermioncirq/trotter/simulate_trotter_test.py]: yes I think that would be more appropriate. Perhaps in the "testing" module.

**Pull request merged**